vms/evm/uptimetracker/uptime_tracker.go

```diff
@@ -15,296 +15,276 @@ import (
 	"github.com/ava-labs/avalanchego/snow/validators"
 	"github.com/ava-labs/avalanchego/utils/set"
 	"github.com/ava-labs/avalanchego/utils/timer/mockable"
+	"errors"
 )
 
-// Validator represents a validator in the state
-type Validator struct {
-	ValidationID   ids.ID     `json:"validationID"`   // Unique validation identifier
-	NodeID         ids.NodeID `json:"nodeID"`         // Node identifier
-	Weight         uint64     `json:"weight"`         // Validator weight/stake
-	StartTimestamp uint64     `json:"startTimestamp"` // When validation started
-	IsActive       bool       `json:"isActive"`       // Whether validator is currently active
-	IsL1Validator  bool       `json:"isL1Validator"`  // Whether this is an L1 validator
-}
+var errNotFound = errors.New("not found")
 
-// UptimeTracker maintains local validator state synchronized with the P-Chain validator set.
-// It tracks validator uptime and manages validator lifecycle events (additions, updates, removals)
-// for the EVM subnet.
-// TODO: consider adding a metrics object to the UptimeTracker
+// UptimeTracker tracks uptime information for validators
 type UptimeTracker struct {
 	validatorState validators.State
 	subnetID       ids.ID
-	// updatedData tracks the updates since WriteValidator was last called
 	manager    uptime.Manager
-	pausedVdrs set.Set[ids.NodeID]
+	clock *mockable.Clock
+
+	lock sync.Mutex
+	height int
 	state      *state
-	// connectedVdrs is a set of nodes that are connected to the manager.
-	// This is used to immediately connect nodes when they are unpaused.
-	connectedVdrs set.Set[ids.NodeID]
-	clock         *mockable.Clock
+	synced              bool
+	connectedValidators set.Set[ids.NodeID]
+	// Deactivated validators are treated as being offline
+	deactivatedValidators set.Set[ids.NodeID]
 }
 
-// NewUptimeTracker returns a new validator state
-// that manages the validator state and the uptime manager.
-// ValidatorState is not thread safe and should be used with the VM locked.
-func NewUptimeTracker(
+// New returns a new instance of UptimeTracker
+func New(
 	validatorState validators.State,
 	subnetID ids.ID,
 	db database.Database,
 	clock *mockable.Clock,
 ) (*UptimeTracker, error) {
-	state, err := newState(db)
+	s, err := newState(db)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create state: %w", err)
+		return nil, fmt.Errorf("failed to initialize state: %w", err)
 	}
 
-	// Create the UptimeTracker first
 	return &UptimeTracker{
 		validatorState: validatorState,
 		subnetID:       subnetID,
-		manager:        uptime.NewManager(state, clock),
-		pausedVdrs:     make(set.Set[ids.NodeID]),
-		state:          state,
-		connectedVdrs:  make(set.Set[ids.NodeID]),
-		clock:          clock,
+		manager: uptime.NewManager(s, clock),
+		clock:  clock,
+		height: -1,
+		state:   s,
 	}, nil
 }
 
-// GetValidator returns the validator data for the given validationID
-func (u *UptimeTracker) GetValidator(vID ids.ID) (Validator, bool) {
```

Comment:
This looks confusing - I'm getting rid of all of the Get* apis. I don't think we need them - EVM already knows about the P-Chain, so we could just grab the validator set from there, and get the uptimes from `UptimeTracker`.